Orange/statistics/util.py

```diff
@@ -160,8 +160,8 @@ def stats(X, weights=None, compute_variance=False):
         If the length of the weight vector does not match the length of the
         array
     """
-    if weights is not None:
-        X = X * weights
+    if weights is not None and X.dtype != object:
+        X = X * weights[:, None]
```

Comment:
Use `np.c_[weights]`. It will ensure it's a column as opposed to raising a dimension (which won't work if weights is already passed as column).